homeassistant/components/yeelight/light.py

```diff
@@ -135,13 +135,28 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     device = hass.data[DATA_YEELIGHT][discovery_info[CONF_HOST]]
     _LOGGER.debug("Adding %s", device.name)
 
+    model = device.model
     custom_effects = _parse_custom_effects(discovery_info[CONF_CUSTOM_EFFECTS])
 
-    lights = [YeelightLight(device, custom_effects=custom_effects)]
-
-    if device.is_ambilight_supported:
-        lights.append(
-            YeelightAmbientLight(device, custom_effects=custom_effects))
+    lights = []
+
+    def _lights_setup_helper(klass):
+        lights.append(klass(device, custom_effects=custom_effects))
+
+    if model in ('mono', 'mono1') or device.type == BulbType.White:
```

Comment:
I don't really like mixing `model` and `device.type` comparison like this, as it makes reading hard. But it's not a showstopper.

Reply:
Good point. I will work on splitting it into 2 methods, when model is declared or fetched via device.type

Reply:
Its simplified now. Please take a look